src/server.rs

```diff
@@ -12,48 +29,211 @@ pub struct ServerConfig {
     pub address: SocketAddr,
 }
 
-pub async fn load_server_config(
-    config: &std::sync::Arc<StewardConfig>,
-) -> Result<ServerConfig, Error> {
-    let tls_config = load_tls_config(
-        &config.server.server_cert_path,
-        &config.server.server_key_path,
-        config.server.client_ca_cert_path.clone(),
-    )
-    .await?;
+pub(crate) async fn start_server_management_thread(mut rx: Receiver<()>) {
+    // wait for the cache to be initialized
+    rx.recv().await;
+
+    loop {
+        let cancellation_token = CancellationToken::new();
+        let server_handle = tokio::spawn(start_server(cancellation_token.clone()));
```

Comment:
I assume you tested this, but does the cancellation token work if you clone it rather than pass a reference to the existing one?

Reply:
Yes, it's designed to cancel all clones of the original when one is cancelled
https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html#method.clone